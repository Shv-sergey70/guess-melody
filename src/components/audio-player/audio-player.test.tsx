import * as React from 'react';
import {configure, mount} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import AudioPlayer from "./audio-player";

configure({adapter: new Adapter()});

describe(`AudioPlayer correctly renders`, () => {
  const pathToMusic = `my/custom/way/to/music.mp3`;

  test(`Play button is disabled and has play class`, () => {
    const tree = mount(
        <AudioPlayer
          isPlaying={false}
          isLoading={true}
          changePlayingState={jest.fn()}>
          <audio src={pathToMusic}/>
        </AudioPlayer>
    );

    expect(tree).toMatchSnapshot();
  });

  test(`Play button is enabled and has pause class`, () => {
    const tree = mount(
        <AudioPlayer
          isPlaying={true}
          isLoading={false}
          changePlayingState={jest.fn()}>
          <audio src={pathToMusic}/>
        </AudioPlayer>
    );

    expect(tree).toMatchSnapshot();
  });
});
