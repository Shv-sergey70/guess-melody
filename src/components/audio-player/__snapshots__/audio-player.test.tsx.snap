// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer correctly renders Play button is disabled and has play class 1`] = `
<AudioPlayer
  changePlayingState={[MockFunction]}
  isLoading={true}
  isPlaying={false}
>
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[MockFunction]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio
      src="my/custom/way/to/music.mp3"
    />
  </div>
</AudioPlayer>
`;

exports[`AudioPlayer correctly renders Play button is enabled and has pause class 1`] = `
<AudioPlayer
  changePlayingState={[MockFunction]}
  isLoading={false}
  isPlaying={true}
>
  <button
    className="track__button track__button--pause"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio
      src="my/custom/way/to/music.mp3"
    />
  </div>
</AudioPlayer>
`;
